piwarsimulator.py

```diff
@@ -49,11 +51,24 @@
 running = True  # state of simulator
 ctrlc_count = 0  # if hitting 3 try and sys.exit
 
+class OperationMode(Enum):
+    Simulation = "simulation"
+    SensorSimulation = "sensor_simulation"
+    Control = "control"
+    ControlSimulation = "control_simulation"
+    EverythingSimButVision = "everything_sim_but_vision"
+    def __str__(self):
+        return self.value
+
+class OmnicamConnectionMode(Enum):
```

Comment:
An Enum is a much more readable and modular solution - depending on configuration, we might end up with more than two options here.

Reply:
I can't think how else it would connect, but OK. If we're collecting a lot of miscellaneous small Enums they let's dump them in util so we know where to find them.

Reply:
If it's not just local maybe that needs to be a text field to enter an IP address rather than an Enum?

Reply:
I'd left it in the piwarsimulator module since it's only necessary for the classes in it (also my reasoning for agreeing that we should return the LIST_ stuff). The way it's being used, it, and the other Enum, are uniquely designed to make the parsing of arg switches easier and serve no purpose outside of the the piwarsim scope.

I had considered making it part of the Vision360 class (which is where I'd have put it if it was a pure internal data class), but seeing as it's directly related to user input in the main program, and that user input can change things that are outside of the scope of the constructor of the Vision360 class, I think it's more closely associated to here. For instance, if I wanted to make an additional mode that spoofed camera data to the simulator via the Vision360 client class, that'd need to be setup and configured in the main class (and would also require it's own mode - maybe something like `local_with_simulated_camera`). In this way, this Enum can act as a comprehensive switch without getting tangled up in representing any actual data.

It could live anywhere, but that's just my rationale for having it there instead. I almost think it'd be saner to move the LIST_ stuff back into piwarsim too because it also only gets used by piwarsim (although, that does feel like it has the range to be used outside of it - it's not just a convenience class for argparsing) rather than move things out, but if you want all the Enums moved into util, then we can do that too.

Reply:
As above I wrote above, I understand the logic now. Happy to shuffle around as needed.